T30304 sdk element tests

dylanmccall opened the pull request:
This moves most of the logic for running tests for SDK components from [eos-build](https://github.com/endlessm/eos-build/blob/master/ci-integration/jenkins-jobs/sdk.yaml#L104) to here.

Some benefits to this approach:

- We can remove a bunch of project-specific code from Jenkins. All it needs to know is what the test element is called for a particular SDK component's repository, and the rest is managed here.
- By running tests via buildstream, we are able to have a standardized test environment that is the same on Jenkins as on someone's development system.
- Test runs are in the best case very fast: buildstream will reuse build artifacts where possible.

Some disadvantages:

- Test elements here are usually _duplicates_ of the originals (using buildstream's file include syntax), rather than _dependants_. This means BuildStream will always need to rebuild a test element separately from the canonical element if it has changed, and to run tests against a local git checkout you need to open a workspace for the test element rather than the canonical one. On the other hand, this means we no longer need the eos_tests variable which was used to generate a special build of certain components with the appropriate test configuration.

dylanmccall committed:
tests: Add xvfb-run

This allows us to run tests inside a buildstream shell without requiring
additional setup on the host.

https://phabricator.endlessm.com/T30304

dylanmccall committed:
tests: Add test runners for Endless SDK components

https://phabricator.endlessm.com/T30304

dylanmccall committed:
project: Remove eos_tests variable

dnohales commented on elements/tests.bst:
```diff
@@ -0,0 +1,14 @@
+- tests/sdk-depends/maxwell.bst
+- tests/sdk-depends/xapian-glib.bst
+# - tests/sdk/eos-knowledge-lib.bst
+# - tests/sdk/eos-metrics.bst
```
Are these comment intended?

dylanmccall replied:
Oh, yeah. I commented those out since they error out at the moment, but shouldn't really be like that :)

The problem with eos-metrics is related to the test environment and resolved in this pull request: https://github.com/endlessm/eos-metrics/pull/42. For eos-knowledge-lib, there's something weird going on which is probably outside the scope of this branch, and once that's fixed tests.bst should run happily!

dnohales commented on elements/tests-build-depends/xauth.bst:
```diff
@@ -0,0 +1,20 @@
+  - xauth-*
+  exclude:
+  - xauth-*.99.*
+  ref: xauth-1.1-5-gaaf037ec5c576e46318935feaf6e2b7407ff11a0
```
Missing trailing new line.

dylanmccall replied:
Fixed. Thanks!

dnohales commented on elements/sdk-depends/eos-shard.bst:
```diff
@@ -2,6 +2,7 @@ kind: autotools
 
 build-depends:
 - freedesktop-sdk.bst:public-stacks/buildsystem-autotools.bst
+- sdk-build-depends/jasmine-gjs.bst
```
Couldn't you "List Append" this in `elements/tests/sdk/libdmodel.bst`?

dylanmccall replied:
Oh yeah, that would be better :) There are a couple of modules (eos-sdk and eos-knowledge-content-renderer) which refuse to build if jasmine isn't available: https://github.com/endlessm/eos-knowledge-content-renderer/blob/master/tests/meson.build#L7. Probably fixable, but for now I'll just leave jasmine-gjs.bst as a build dependency for regular builds over there.

dnohales reviewed (changes requested):
Awesome. Way cleaner now, left some comments.

dylanmccall committed:
tests: Add special dbus configuration for eos-metrics

This changes dbus configuration in the test environment to allow
anonymous authentication, so we can easily run dbus-daemon without
adding extra stuff to the sandbox.

https://phabricator.endlessm.com/T30304

dylanmccall committed:
tests: Add jasmine-gjs as a tests build dependency

Remove this as a regular build dependency where possible since it isn't
strictly needed. However, the build system for some components fails if
the `jasmine` program is not available, so it is left as a build
dependency for those elements.

dylanmccall committed:
tests: Uncomment all dependencies in tests.bst

Some of these are still failing to build, but it is clearer if these are
not commented.

dnohales reviewed (approved):
This looks good for me, please merge it when ready, I suppose this will require changes in eos-build
